File: demos/ios/Demos/Demos/views/HistogramBarChart.m

#import "HistogramBarChart.h"

@implementation HistogramBarChart

- (void)chartRender {
    NSString *jsonPath = [[NSBundle mainBundle] pathForResource:@"Res/mockData_setionInterval" ofType:@"json"];
    NSString *jsonData = [NSString stringWithContentsOfFile:jsonPath encoding:NSUTF8StringEncoding error:nil];
    self.chart.canvas(self.canvasView).padding(20, 10, 20, 0.f);
    self.chart.source([F2Utils toJsonArray:jsonData]);
    self.chart.axis(@"x", @{@"grid": @(NO)});
    self.chart.axis(@"y", @{@"grid": @{@"type": @"dash", @"lineWidth": @(2), @"dash": @[@(6), @(2)]}});
    self.chart.scale(@"x", @{@"range": @[@(0.1), @(0.9)]});
    self.chart.scale(@"y", @{@"nice": @(YES)});
    self.chart.interval().position(@"x*y");
    self.chart.render();
}

@end

File: demos/ios/Demos-Swift/Demos-Swift/Views/HistogramBarChart.swift
//
//  HistogramBarChart.swift
//  F2Native
//
//  Created by Westin on 2024-06-09.
//

import F2

class HistogramBarChart: BaseLineChart {
    
    override func chartRender() {
        guard let jsonPath = Bundle.main.path(forResource: "Res/mockData_setionInterval", ofType: "json"),
              let jsonDataString = try? String(contentsOfFile: jsonPath),
              let canvasView = self.canvasView,
              let chart = self.chart else {
            return
        }
        let jsonData = F2Utils.toJsonArray(jsonDataString)
        chart.canvas()(canvasView).padding()(20, 10, 20, 0)
        chart.source()(jsonData)
        chart.axis()("x", ["grid": false])
        chart.axis()("y", ["grid": ["type": "dash", "lineWidth": 2, "lineDash": [6, 2]]])
        chart.scale()("x", ["range": [0.1, 0.9]])
        chart.scale()("y", ["nice": true])
        chart.interval()().position()("x*y")
        chart.animate()(true)
        chart.render()()
    }
}
